package-res/resources/web/pentaho/util/arg.js

```diff
@@ -20,23 +20,73 @@ define([
 
   var A_slice = Array.prototype.slice;
 
+  /**
+   * @name pentaho.util.arg
+   * @class
+   * @amd pentaho/util/arg
+   *
+   * @classDesc Util methods for keyword args.
+   *
+   * The AMD module returns the arg utils.
+   *
+   * @description Performs actions on keyword args.
+   */
   return {
+    /**
+     * Provides the optional property value of a given object or the given default value if the given object does
+     * not contain the sought after property value.
+     *
+     * @param {?object} [obj] The object to search for given propName param.
+     * @param {String} [propName] Property name to find in obj param.
+     * @param {any} [defaultValue] The default value to return if propName param is not found. The default value of this param is undefined and is considered specified when its value is not a nully value. {@link Nully}
+     *
```

Comment:
ok, I see now, the [] imply the param is optional. What does the {?object} imply?